File: examples/java/HelloWorld.java
/**
 * Simple Java program for smoke-testing the Java debug adapter.
 *
 * Compile:  javac HelloWorld.java
 * Run:      java HelloWorld
 */
public class HelloWorld {

    static int add(int a, int b) {
        int result = a + b;     // line 10
        return result;          // line 11
    }

    static String greet(String name) {
        String greeting = "Hello, " + name + "!";  // line 15
        return greeting;                             // line 16
    }

    public static void main(String[] args) {
        int x = 10;                        // line 20
        int y = 20;                        // line 21
        int sum = add(x, y);               // line 22
        String msg = greet("World");       // line 23
        System.out.println(msg);           // line 24
        System.out.println("Sum: " + sum); // line 25
    }
}

File: examples/cpp/hello_world.cpp
// Simple C++ example for mcp-debugger smoke tests (issue #325).
// Compile: g++ -gdwarf-4 -O0 -o hello_world hello_world.cpp
#include <iostream>
#include <string>
#include <vector>

// Function-breakpoint target: set a function breakpoint on "compute_answer".
int compute_answer(int base, int factor) {
    int result = base * factor;
    return result + 2;
}

int main() {
    int count = 10;
    std::string greeting = "Hello from C++";  // asserted verbatim by tests/e2e/mcp-server-smoke-cpp.test.ts; line numbers here are breakpoint targets -- do not shift
    std::vector<int> values = {1, 2, 3};
    int answer = compute_answer(count, 4);
    std::cout << "CPP_DEBUG_MARKER: " << greeting << " answer=" << answer << std::endl;
    std::cout << "values has " << values.size() << " elements" << std::endl;
    return 0;
}
